File: database/src/main/java/com/example/database/dtos/UserDTO.java
package com.example.database.dtos;

import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

import java.time.Instant;
import java.util.UUID;

@Value
@Builder
@Jacksonized
public class UserDTO {
    UUID userId;

    String username;

    String firstName;

    String lastName;

    String email;

    String about;

    Integer matchesPlayed;

    Integer matchesWon;

    Instant joinedAt;
}

File: MyApplication/app/src/main/java/com/example/myapplication/dtos/UserDTO.kt
package com.example.myapplication.dtos

import java.util.Date
import java.util.UUID

data class UserDTO(
    val userId: UUID,
    val username: String,
    val firstName: String,
    val lastName: String,
    val email: String,
    val about: String,
    val matchesPlayed: Int,
    val matchesWon: Int,
    val joinedAt: Date
)